### PR TITLE
Enables linkit on link fields in various entity form displays.

### PR DESCRIPTION
**Pull request by neilblair**

### CODE REVIEW

File: config/sync/core.entity_form_display.node.news.default.yml [hunk 1]
```diff
@@ -72,11 +72,12 @@ content:
     settings:
       placeholder_url: ''
       placeholder_title: ''
+      linkit_profile: default
     third_party_settings: {  }
-    type: link_default
+    type: linkit
     region: content
   field_summary:
-    type: text_textarea
```
Comment (neilblair) [inline config/sync/core.entity_form_display.node.news.default.yml:79]: Not sure why field_summary is being changed from text_textarea to string_textarea here - but I think string_textarea is correct

File: config/sync/core.entity_form_display.node.news.default.yml [hunk 2]
```diff
@@ -72,11 +72,12 @@ content:
...
+    type: string_textarea
```
Comment (neilblair) [inline config/sync/core.entity_form_display.node.news.default.yml:80]: Not sure why field_summary is being changed from text_textarea to string_textarea here - but I think string_textarea is correct

Comment (johangant) [reply config/sync/core.entity_form_display.node.news.default.yml:80]: Not sure either, but if it's autogenerated by config export it's entirely possible that's a change from core itself.

**Review by johangant — APPROVED** [review]
LGTM. LinkIt taking over :)

**Pull request merged**